src/webgpu/web_platform/copyToTexture/image.spec.ts

```diff
@@ -146,6 +146,79 @@ g.test('from_image')
     );
   });
 
+g.test('from_fully_transparent_image')
+  .desc(
+    `
+  Test HTMLImageElement with alpha 0 can be copied to WebGPU texture correctly.
+
+  Use a prebaked 2x2 fully transparent image as source.
+
+  Then call copyExternalImageToTexture() to do a copy to the 0 mipLevel of dst texture,
+  and read the contents out to compare with the HTMLImageElement contents.
+  When dest alpha mode is:
+  - premultiplied, the content should be (0, 0, 0, 0)
+  - not premultiplied, the content should be the same as prebaked
+    pixel values (255, 102, 153, 0).
+
+  The tests covers:
+  - Source HTMLImageElement is fully transparent with valid dest alphaMode.
+  And the expected results are all passed.
+  `
+  )
+  .params(u => u.combine('dstPremultiplied', [true, false]))
+  .beforeAllSubcases(t => {
+    if (typeof HTMLImageElement === 'undefined') t.skip('HTMLImageElement not available');
+  })
+  .fn(async t => {
+    const { dstPremultiplied } = t.params;
+
+    const kColorFormat = 'rgba8unorm';
+    const kImageWidth = 2;
+    const kImageHeight = 2;
+
+    const imageCanvas = document.createElement('canvas');
+    imageCanvas.width = kImageWidth;
+    imageCanvas.height = kImageHeight;
+
+    // Prebaked fully transparent image with content (255, 102, 153, 0)
+    const image = new Image(kImageWidth, kImageHeight);
```

Comment:
Just wondering, do we have tests for partially-transparent images as well? I don't see any here.

If not, perhaps file a bug for this as well.

Reply:
No, I don't think we have partially-transparent image test coverage(And only ImageBitmap has partially-transparent coverage). File issue 3668